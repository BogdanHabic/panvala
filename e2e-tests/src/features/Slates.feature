@MetaMask_Local
Feature: Slates

Background: Slate Submission Stage
  Given I navigate to the "Slates" page
  And The Panvala "Slates" page is displayed
  And I Time Travel to the "Slate Submission" stage in Panvala
  And The Panvala "Slates" page is displayed

Scenario: Create a Grant Slate
  When I click add slates on the Panvala slates page
  And The Panvala "Create Slates" page is displayed
  And I select the Grant Slate slate type on the Panvala Create a Slate page
  And The Panvala "Create Grant" page is displayed
  And I enter the slate details on the Panvala Create a Grant Slate page
  And I click the Create Slate button on the Panvala Create Grant Slate page
  Then The "Slate submitted." modal dialog is displayed reading
  """
  Now that your slate has been created you and others have the ability to stake tokens on it to propose it to token holders. Once there are tokens staked on the slate it will be eligible for a vote.
  """

Scenario: Create a Governance Slate
  When I click add slates on the Panvala slates page
  And The Panvala "Create Slates" page is displayed
  And I select the Governance Slate slate type on the Panvala Create a Slate page
  And The Panvala "Create Governance" page is displayed
  And I enter the slate details on the Panvala Create a Governance Slate page
  And I click the Create Slate button on the Panvala Create Governance Slate page
  Then The "Slate submitted." modal dialog is displayed reading
  """
  Now that your slate has been created you and others have the ability to stake tokens on it to propose it to token holders. Once there are tokens staked on the slate it will be eligible for a vote.
  """

@Ignore
Scenario: View a Grant Slate
  When I click add slates on the Panvala slates page
  And The Panvala "Create Slates" page is displayed
  And I select the Grant Slate slate type on the Panvala Create a Slate page
  And The Panvala "Create Grant" page is displayed
  And I enter the email "peter.yinusa@gmail.com" on the Panvala Create Grant Slate page
  And I enter the first name "Peter" on the Panvala Create Grant Slate page
  And I enter the description "Description" on the Panvala Create Grant Slate page
  And I click the Recommend no action option on the Panvala Create Grant Slate page
  And I click the Stake no option on the Panvala Create Grant Slate page
  And I click the Create Slate button on the Panvala Create Grant Slate page
  Then The "Slate submitted." modal dialog is displayed reading
  """
  Now that your slate has been created you and others have the ability to stake tokens on it to propose it to token holders. Once there are tokens staked on the slate it will be eligible for a vote.
  """
  When I accept the modal dialog
  And I click the last grant slate card on the Panvala slates page
  Then The Panvala "Slate" page is displayed
  And The first name is displayed on the Panvala Slate page
  And The description is displayed on the Panvala Slate page

Scenario: View a Governance Slate
  When I click add slates on the Panvala slates page
  And The Panvala "Create Slates" page is displayed
  And I select the Governance Slate slate type on the Panvala Create a Slate page
  And The Panvala "Create Governance" page is displayed
  And I enter the email "peter.yinusa@gmail.com" on the Panvala Create Governance Slate page
  And I enter the first name "Peter" on the Panvala Create Governance Slate page
  And I enter the description "Description" on the Panvala Create Governance Slate page
  And I click the Recommend no action option on the Panvala Create Governance Slate page
  And I click the Stake no option on the Panvala Create Governance Slate page
  And I click the Create Slate button on the Panvala Create Governance Slate page
  Then The "Slate submitted." modal dialog is displayed reading
  """
  Now that your slate has been created you and others have the ability to stake tokens on it to propose it to token holders. Once there are tokens staked on the slate it will be eligible for a vote.
  """
  When I accept the modal dialog
  Then The Panvala "Slates" page is displayed
  When I click the last governance slate card on the Panvala slates page
  Then The Panvala "Slate" page is displayed
  And The first name is displayed on the Panvala Slate page
  And The description is displayed on the Panvala Slate page