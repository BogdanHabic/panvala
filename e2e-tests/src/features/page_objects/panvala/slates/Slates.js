import { BasePanvala } from '../BasePanvala';
import { HtmlElement } from '../../../framework/controls/HtmlElement';

class Slates extends BasePanvala {

  constructor() {
    super('/slates');
    this.addSlate = () => new HtmlElement('a[href="/slates/create"]');
    this.deadline = () => new HtmlElement('.Deadline__StyledDeadline-sc-1a3aqlz-0');
    this.currentTab = () => new HtmlElement('.VisibilityFilter__VisibilityFilterContainer-sc-1b1hgnt-0 div:nth-child(1)');
    this.pastTab = () => new HtmlElement('.VisibilityFilter__VisibilityFilterContainer-sc-1b1hgnt-0 div:nth-child(2)');
  }

  clickAddSlate() {
    return this.addSlate().click();
  }

  getDeadline() {
    return this.deadline().getText();
  }

  async clickTab(tab) {
    let tabElement;
    switch (tab) {
      case 'past':
        tabElement = this.pastTab();
        break;
      case 'current':
        tabElement = this.currentTab();
        break;
      default:
        throw new Error(`Please specify a valid tab, unknown tab $(tab)`);
    }
    return tabElement.click();
  }

}

export { Slates };
