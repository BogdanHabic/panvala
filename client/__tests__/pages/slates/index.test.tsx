import * as React from 'react';
import { render, cleanup } from 'react-testing-library';
import 'jest-dom/extend-expect';

import Slates from '../../../pages/index';

afterEach(cleanup);

describe.skip('Slates', () => {
  it('should render correctly Slates component', () => {
    const slates = render(<Slates />);

    expect(slates).toMatchSnapshot();
  });
});
