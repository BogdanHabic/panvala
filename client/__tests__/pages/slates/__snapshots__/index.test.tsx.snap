// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slates should render correctly Slates component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <div
            class="flex justify-between"
          >
            <div
              class="flex"
            >
              <h1
                class="mr3 RouteTitle-ehvjdd-0 fHYele"
              >
                Slates
              </h1>
              <a
                class="link flex items-center"
                href="/slates/create"
              >
                <div
                  class="Button__Wrapper-sc-1vedn6s-2 jHiEij"
                >
                  <button
                    class="Button__StyledButton-sc-1vedn6s-0 izJlFT"
                    type="default"
                  >
                    Add Slate
                  </button>
                </div>
              </a>
            </div>
            <div
              class="Deadline__Wrapper-sc-1a3aqlz-1 gtXKcM"
            >
              <div
                class="Tag__Wrapper-l3n6sk-1 ceKKQx"
              >
                <div
                  class="Deadline__StyledDeadline-sc-1a3aqlz-0 jpSHwD Tag__StyledTag-l3n6sk-0 llOOnS"
                >
                  SLATE STAKING DEADLINE 1969-12-31 AT 19:00 PM EST
                </div>
              </div>
            </div>
          </div>
          <div
            class="slates__VisibilityFilterContainer-sc-15h3ej7-0 hVkzXI"
          >
            <div
              class="Button__Wrapper-sc-1vedn6s-2 jHiEij"
            >
              <button
                class="Button__StyledButton-sc-1vedn6s-0 eXurug"
              >
                All
              </button>
            </div>
            <div
              class="Button__Wrapper-sc-1vedn6s-2 jHiEij"
            >
              <button
                class="Button__StyledButton-sc-1vedn6s-0 lahyEh"
              >
                Current
              </button>
            </div>
            <div
              class="Button__Wrapper-sc-1vedn6s-2 jHiEij"
            >
              <button
                class="Button__StyledButton-sc-1vedn6s-0 lahyEh"
              >
                Past
              </button>
            </div>
          </div>
          <div
            class="slates__CardsWrapper-sc-15h3ej7-1 gPFIfm"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <div
          class="flex justify-between"
        >
          <div
            class="flex"
          >
            <h1
              class="mr3 RouteTitle-ehvjdd-0 fHYele"
            >
              Slates
            </h1>
            <a
              class="link flex items-center"
              href="/slates/create"
            >
              <div
                class="Button__Wrapper-sc-1vedn6s-2 jHiEij"
              >
                <button
                  class="Button__StyledButton-sc-1vedn6s-0 izJlFT"
                  type="default"
                >
                  Add Slate
                </button>
              </div>
            </a>
          </div>
          <div
            class="Deadline__Wrapper-sc-1a3aqlz-1 gtXKcM"
          >
            <div
              class="Tag__Wrapper-l3n6sk-1 ceKKQx"
            >
              <div
                class="Deadline__StyledDeadline-sc-1a3aqlz-0 jpSHwD Tag__StyledTag-l3n6sk-0 llOOnS"
              >
                SLATE STAKING DEADLINE 1969-12-31 AT 19:00 PM EST
              </div>
            </div>
          </div>
        </div>
        <div
          class="slates__VisibilityFilterContainer-sc-15h3ej7-0 hVkzXI"
        >
          <div
            class="Button__Wrapper-sc-1vedn6s-2 jHiEij"
          >
            <button
              class="Button__StyledButton-sc-1vedn6s-0 eXurug"
            >
              All
            </button>
          </div>
          <div
            class="Button__Wrapper-sc-1vedn6s-2 jHiEij"
          >
            <button
              class="Button__StyledButton-sc-1vedn6s-0 lahyEh"
            >
              Current
            </button>
          </div>
          <div
            class="Button__Wrapper-sc-1vedn6s-2 jHiEij"
          >
            <button
              class="Button__StyledButton-sc-1vedn6s-0 lahyEh"
            >
              Past
            </button>
          </div>
        </div>
        <div
          class="slates__CardsWrapper-sc-15h3ej7-1 gPFIfm"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
